### PR TITLE
Implement excluding sensitive data from logs

### PR DESCRIPTION
**Pull request by nechaido**
Co-authored-by: Mykola Bilochub <nbelochub@gmail.com>

### CODE REVIEW

File: lib/log.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+const removeSensitiveData = (definition, instance) => {
+  const filtered = {};
+  for (const name of Object.keys(instance)) {
+    const prop = definition[name];
+    if (
```
Comment (tshemsedinov) [inline lib/log.js]:
```suggestion
    if (prop) {
      if (prop.sensitive) continue;
      if (prop.definition && prop.definition.sensitive) continue;
    }
```

Comment (tshemsedinov) [reply lib/log.js]: + remove next 3 lines and put `filtered[name] = instance[name];` after if statement

File: lib/log.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      : ms.actions.get(action);
+    if (record.Query) {
+      record.Query = removeSensitiveData(schema.definition.Args, record.Query);
+    }
+    if (Array.isArray(record.Response)) {
+      record.Response = [
+        removeSensitiveData(schema.definition.Returns, record.Response[0]),
+      ];
+    }
+  } else if (operation === 'Read') {
```
Comment (tshemsedinov) [inline lib/log.js]:
```suggestion
    return;
  }
  if (operation === 'Read') {
```

File: lib/log.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    let schema = ms.categories.get(category);
+    if (record.getDetails) schema = schema.definition[record.getDetails];
+    if (Array.isArray(record.Response)) {
+      record.Response = record.Response.map(record =>
+        removeSensitiveData(schema.definition, record)
+      );
+    } else if (record.Response) {
+      record.Response = removeSensitiveData(schema.definition, record.Response);
+    }
+  } else {
```
Comment (tshemsedinov) [inline lib/log.js]:
```suggestion
    return;
  }
```
and remove indentation for next block

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/globalstorage/commit/2efdade92fd943ceb1bd3d92a754b9a3a641b7fb.

**Pull request closed without merging**